charts/pravega/templates/post-install-hooks.yaml

```diff
@@ -0,0 +1,119 @@
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+rules:
+- apiGroups:
+  - pravega.pravega.io
+  resources:
+  - "*"
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - "pods"
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - "pods/exec"
+  verbs:
+  - create
+---
+
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+subjects:
+- kind: ServiceAccount
+  name: {{ template "hook.fullname" . }}
+  namespace: "{{.Release.Namespace}}"
+roleRef:
+  kind: Role
+  name: {{ template "hook.fullname" . }}
+  apiGroup: rbac.authorization.k8s.io
+
+---
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+      "helm.sh/hook": post-install
+      "helm.sh/hook-weight": "1"
+      "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+data:
+  validations.sh: |
+    #!/bin/sh
+    set -e
+    sleep 30
+    kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.conditions[?(@.type=="PodsReady")].status}' | grep True
+    currentReplicas=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.currentReplicas}'`
+    readyReplicas=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.readyReplicas}'`
+    currentVersion=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.currentVersion}'`
+    targetVersion=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.spec.version}'`
+    echo "PravegaCluster replicas: $currentReplicas"
+    echo "PravegaCluster readyReplicas: $readyReplicas"
+    echo "PravegaCluster currentVersion: $currentVersion"
+    echo "PravegaCluster targetVersion: $targetVersion"
+    if [ $currentReplicas != $readyReplicas ]; then
+        exit 1
+    fi
+    if [ $currentVersion != $targetVersion ]; then
+        exit 2
+    fi
+---
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "2"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+spec:
+  backoffLimit: 3
```

Comment:
My thoughts are that we increase the number of retries, the `helm install`  command will keep waiting for that long (30 * number_of_retries) for before giving any output to the user in case there was a failure. Increasing the number of retries might not help in cases where the failure happens due to a generic reason like `ImagePullBackOff`.

Reply:
Ok makes sense -- we can always retweak if need-be!